docs/proposals/20210203-unmanaged-infrastructure.md

```diff
@@ -0,0 +1,236 @@
+
+#### Story 2 - Restricted access to cloud provider APIs
+As a cluster operator I want to use CAPI to orchestrate kubernetes bootstraping while restricting the privileges I need to grant for my cloud provider because of organisational cloud security constraints.
+
+#### Story 3 - Consuming existing cloud infrastructure
+As a cluster operator I want to use CAPI to orchestate Kubernetes bootstraping while reusing infrastructure that has already been created in the organisation either by me or another team.
+
+### Implementation Details/Notes/Constraints
+
+**Managed**
+
+- It will be default and will preserve existing behaviour. An InfraCluster CR without the `cluster.x-k8s.io/managed-by: "<name-of-system>"` annotation.
+
+
+**Externally Managed**
+
+An InfraCluster CR with the `cluster.x-k8s.io/managed-by: "<name-of-system>"` annotation.
+
+The provider InfraCluster controller must:
+- Skip any reconciliation of the resource.
+
+- Not update the resource or its status in any way
+
+The external management system must:
+
+- Populate all required fields within the InfraCluster spec to allow other CAPI components to continue as normal.
+
+- Adhere to all Cluster API contracts for infrastructure providers.
+
+- When the infrastructure is ready, set the appropriate status as is done by the provider controller today.
+
+#### Provider implementation changes
+
+To enable providers to implement the changes required by this contract, Cluster API is going to provide a new `predicates.ResourceExternallyManaged` predicate as part of its utils.
+
+This predicate filters out any resource that has been marked as "externally managed" and prevents the controller from reconciling the resource.
+
+### Security Model
+
+When externally managed, the required cloud provider privileges are significantly reduced when compared with a traditionally managed cluster.
+The only privileges required by CAPI are those that are required to manage machines.
+
+For example, when an AWS cluster is managed by CAPI, persmissions are required to be able to create VPCs and other networking components that are managed by the AWSCluster controller. When externally managed, these permissions are not required as the external entity is reponsible for creating such components.
+
+### Risks and Mitigations
+
+#### What happens when a user converts an externally managed InfraCluster to a managed InfraCluster?
+
+Whether the flag for being managed is an annotation or field on the InfraCluster, there currently is no immutability support for CRD fields or annotations within the Kubernetes API.
+
+This means that, once a user has created their externally managed InfraCluster, they could at some point, update the annotation/field to make the InfraCluster appear to be managed.
+
+There is no way to predict what would happen in this scenario.
```

Comment:
what about API conversions/downgrades? If a user creates a v1alpha4 externally managed cluster and converts it to v1alpha3 for example (v1alpha3 having no knowledge of this annotation)

Reply:
mm I think we should communicate clearly this is supported >= v1alpha4. Do you have any suggestion to mitigate the this scenario?